test/server/cards/01_SOR/IdenVersioInfernoSquadCommander.spec.js

```diff
@@ -0,0 +1,92 @@
+describe('Iden Version, Inferno Squad Commander', function() {
+    integration(function() {
+        describe('Iden\'s undeployed ability', function() {
+            beforeEach(function () {
+                this.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['wampa', 'battlefield-marine'],
+                        leader: 'iden-versio#inferno-squad-commander',
+                        base: { card: 'dagobah-swamp', damage: 5 }
+                    },
+                    player2: {
+                        groundArena: ['specforce-soldier', 'atst'],
+                    }
+                });
+            });
+
+            it('should heal 1 from if an opponent\'s unit was defeated this phase', function () {
+                // case 1: nothing happens, no cards defeated
+                this.player1.clickCard(this.idenVersio);
+                this.player1.clickPrompt('Heal 1 from base if an opponent\'s unit was defeated this phase');
+                expect(this.idenVersio.exhausted).toBe(true);
+                expect(this.p1Base.damage).toBe(5);
+
+                this.idenVersio.exhausted = false;
+                this.player2.passAction();
+
+                // case 2: friendly unit defeated
+                this.player1.clickCard(this.battlefieldMarine);
+                this.player1.clickCard(this.atst);
+                this.player2.passAction();
+
+                this.player1.clickCard(this.idenVersio);
+                this.player1.clickPrompt('Heal 1 from base if an opponent\'s unit was defeated this phase');
+                expect(this.idenVersio.exhausted).toBe(true);
+                expect(this.p1Base.damage).toBe(5);
+
+                this.idenVersio.exhausted = false;
+                this.player2.passAction();
+
+                // case 3: enemy unit defeated
```

Comment:
I feel like each of these cases should be its own unit test. I know we've been consolidating them sometimes but I think we might want to start making our tests more granular

Reply:
The problem is that our tests are very heavy and take a lot of time since to set up, so there would be a concrete impact if we suddenly were adding a lot more tests. Eventually we should look at reducing that but it probably requires state backup tbh (or it might just be that jasmine is slow in general).

I'm particularly aware of it because we have finite Github Actions time, otherwise I'd be more 50/50 about it.

Reply:
E.g., when I tried to run the L5R tests the first time with all of its cards, I think it took 5 minutes.